src/views/checkups/self-validation/components/form/CheckupsSelfValidationFormActions.tsx

```diff
@@ -0,0 +1,147 @@
+import { useAllRunningSelfValidationJobs } from '../hooks/useAllRunningSelfValidationJobs';
+import useCheckupsSelfValidationPermissions from '../hooks/useCheckupsSelfValidationPermissions';
+
+type CheckupsSelfValidationFormActionsProps = {
+  checkupImage: string;
+  isDryRun: boolean;
+  name: string;
+  pvcSize: string;
+  selectedTestSuites: string[];
+  storageCapabilities?: string[];
+  storageClass?: string;
+  testSkips?: string;
+};
+
+const CheckupsSelfValidationFormActions: FC<CheckupsSelfValidationFormActionsProps> = ({
+  checkupImage,
+  isDryRun,
+  name,
+  pvcSize,
+  selectedTestSuites,
+  storageCapabilities,
+  storageClass,
+  testSkips,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const navigate = useNavigate();
+  const [namespace] = useActiveNamespace();
+  const [error, setError] = useState<ReactNode>(null);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const { isPermitted: isCreateSelfValidationPermitted } = useCheckupsSelfValidationPermissions();
+  const [runningSelfValidationJobs] = useAllRunningSelfValidationJobs();
+
+  const selfValidationActionState = useMemo(
+    () =>
+      getActionState(SELF_VALIDATION_ACTION_MODE.RUN, {
+        hasCurrentCheckupRunningJobs: false,
+        hasOtherRunningJobs: false,
+        isCreateSelfValidationPermitted,
+        otherRunningJobs: [],
+        runningSelfValidationJobs: runningSelfValidationJobs || [],
+      }),
+    [isCreateSelfValidationPermitted, runningSelfValidationJobs],
+  );
+
+  const hasRunningCheckup = useMemo(
+    () => selfValidationActionState?.showWarning && !selfValidationActionState?.isEnabled,
+    [selfValidationActionState],
+  );
+
+  const isFormValid = name && checkupImage && selectedTestSuites.length > 0;
+  const isSubmitDisabled =
+    isSubmitting || !isFormValid || hasRunningCheckup || !selfValidationActionState?.isEnabled;
+
+  const showTooltip =
+    hasRunningCheckup &&
+    selfValidationActionState?.showWarning &&
+    selfValidationActionState?.configMapInfo;
+
+  const handleRun = async () => {
+    setError(null);
+    setIsSubmitting(true);
+    try {
+      await createSelfValidationCheckup({
+        checkupImage,
+        isDryRun,
+        name,
+        namespace,
+        pvcSize,
+        selectedTestSuites,
+        storageCapabilities,
+        storageClass,
+        testSkips,
+      });
+      navigate(`/k8s/ns/${namespace}/checkups/${CHECKUP_URLS.SELF_VALIDATION}/${name}`);
+    } catch (e) {
+      kubevirtConsole.error(e);
+      setError(e?.message);
+    } finally {
+      setIsSubmitting(false);
+    }
+  };
+
+  const runButton = (
+    <Button
+      isDisabled={isSubmitDisabled}
+      isLoading={isSubmitting}
+      onClick={handleRun}
+      variant={ButtonVariant.primary}
+    >
+      {t('Run')}
+    </Button>
+  );
+
+  return (
+    <>
+      {error && (
+        <Alert isInline title={t('Error')} variant={AlertVariant.danger}>
```

Comment:
consider using `ErrorAlert` component - it retrieves more information from a standard error